File: Java/Laptop.java
public class Laptop extends Product {
    // Attributes specific to Laptop
    private String processor;
    private int ram;
    private int battery;
    private int warranty;

    // Default constructor
    public Laptop() {
        super(); // call Product() constructor
        this.processor = "";
        this.ram = 0;
        this.battery = 0;
        this.warranty = 0;
    }

    // Constructor with parameters
    public Laptop(int id, String name, String brand, int price,
                  String processor, int ram, int battery, int warranty) {
        super(id, name, brand, price); // call Product constructor with arguments
        this.processor = processor;
        this.ram = ram;
        this.battery = battery;
        this.warranty = warranty;
    }

    // Getters
    public String getProcessor() {
        return processor;
    }

    public int getRam() {
        return ram;
    }

    public int getBattery() {
        return battery;
    }

    public int getWarranty() {
        return warranty;
    }

    // Setters
    public void setProcessor(String processor) {
        this.processor = processor;
    }

    public void setRam(int ram) {
        this.ram = ram;
    }

    public void setBattery(int battery) {
        this.battery = battery;
    }

    public void setWarranty(int warranty) {
        this.warranty = warranty;
    }
}

File: Python/Laptop.py
from Product import Product

class Laptop(Product):
    def __init__(self, id=0, name="", brand="", price=0, processor="", ram=0, battery=0, warranty=0):
        # Call parent constructor (Product) to initialize common attributes
        super().__init__(id, name, brand, price)
        # Private attributes specific to Laptop
        self.__processor = processor
        self.__ram = ram
        self.__battery = battery
        self.__warranty = warranty

    # Getters
    def get_processor(self):
        return self.__processor

    def get_ram(self):
        return self.__ram

    def get_battery(self):
        return self.__battery

    def get_warranty(self):
        return self.__warranty

    # Setters
    def set_processor(self, processor):
        self.__processor = processor

    def set_ram(self, ram):
        self.__ram = ram

    def set_battery(self, battery):
        self.__battery = battery

    def set_warranty(self, warranty):
        self.__warranty = warranty
